.github/instructions/dsc-community-style-guidelines-powershell.instructions.md

```diff
@@ -0,0 +1,159 @@
+- Functions: PascalCase with Verb-Noun format using approved verbs
+- Parameters: PascalCase
+- Variables: camelCase
+- Keywords: lower-case
+- Classes: PascalCase
+- Include scope for script/global/environment variables: `$script:`, `$global:`, `$env:`
+
+## Formatting
+
+### Indentation & Spacing
+- Use 4 spaces (no tabs)
+- One space around operators: `$a = 1 + 2`
+- One space between type and variable: `[String] $name`
+- One space between keyword and parenthesis: `if ($condition)`
+- No spaces on empty lines
+- Try to limit lines to 120 characters
+
+### Braces
+
+- Newline before opening brace (except variable assignments)
+- One newline after opening brace
+- Two newlines after closing brace (one if followed by another brace or continuation)
+
+### Quotes
+
+- Use single quotes unless variable expansion is needed: `'text'` vs `"text $variable"`
+
+### Arrays
+
+- Single line: `@('one', 'two', 'three')`
+- Multi-line: each element on separate line with proper indentation
+- Do not use the unary comma operator (`,`) in return statements to force
+  an array
+
+### Hashtables
+
+- Empty: `@{}`
+- Multi-line: each property on separate line with proper indentation
+- Properties: Use PascalCase
+
+### Comments
+
+- Single line: `# Comment` (capitalized, on own line)
+- Multi-line: `<# Comment #>` format (opening and closing brackets on own line), and indent text
+- No commented-out code
+
+## Functions
+
+- Avoid aliases (use full command names)
+- Avoid `Write-Host` (use `Write-Verbose`, `Write-Information`, etc.)
+- Avoid `Write-Output` (use `return` instead)
+- Avoid `ConvertTo-SecureString -AsPlainText` in production code
+- Don't redefine reserved parameters (Verbose, Debug, etc.)
+- Include a `Force` parameter for functions that support `$PSCmdlet.ShouldContinue` or that use `$PSCmdlet.ShouldProcess`
+- For state-changing functions, use `SupportsShouldProcess`
+- Use `Write-Error` for non-terminating errors
+- Use `Write-Warning` for warnings
+- Use `Write-Debug` for debugging information
+- Use `Write-Verbose` for actionable information
+- Use `Write-Information` for informational messages.
+- Never use backtick as line continuation in production code.
+
+### Structure
+
+```powershell
+<#
+    .SYNOPSIS
+        Brief description
+
+    .DESCRIPTION
+        Detailed description
+
+    .PARAMETER Name
+        Parameter description
+#>
+function Get-Something
+{
+    [CmdletBinding()]
+    [OutputType([String])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter()]
+        [System.String]
+        $OptionalParam
+    )
+
+    # Implementation
+}
+```
+
+### Requirements
+
+- Include `[CmdletBinding()]` on every function
+- Parameter block at top
+- Parameter block: `param ()` if empty, else opening/closing parentheses on own lines
+- Comment-based help: SYNOPSIS, DESCRIPTION (40+ chars), PARAMETER, EXAMPLE sections before function/class
```

Comment:
Add the following:
.INPUTS: type of objects that can be piped to the function. Include a description of the input objects. Repeat this keyword for each input type.

.OUTPUTS: type of the objects that the function returns. Include a description of the returned objects. Repeat this keyword for each output type.